pymongo/network.py

```diff
@@ -297,7 +297,6 @@ def receive_message(
 def wait_for_read(conn: Connection, deadline: Optional[float]) -> None:
     """Block until at least one byte is read, or a timeout, or a cancel."""
     context = conn.cancel_context
-    # Only Monitor connections can be cancelled.
     if context:
```

Comment:
Now that conn.cancel_context is never None, I might consider updating this code block. 

I'm also not crazy about context = conn.cancel_context. It draws attention to a class which appears to be just a boolean. Maybe we could add a docstring to _CancellableContext to explain why it exists.

Reply:
@ShaneHarvey is there a reason `_CancellableContext` can't be removed in favor of a simple boolean?

Reply:
_CancellationContext is implemented via a boolean right now but it's possible to implement it in other ways. For example, at one point it was going to be implemented via a pipe so that the socket read/write could be interrupted more quickly. The other motivation for it being a class was that way a single instance could be shared between multiple connections. I agree the if-statement can be removed if cancel_context is never None. 

I'll leave it up to you if want to remove _CancellableContext and embed the bool within the Connection itself or not.

Reply:
There is a slight complication with removing `_CancellableContext`: `Monitor` takes advantage of `conn.cancel_context` being a reference to an object rather than a bool in order for its `_cancel_context` to affect the underlying connection. If `cancel_context` becomes a bool, `Monitor` will have to hold a reference to the last used connection instead in order to properly cancel the connection when needed.